Lib/test/string_tests.py

```diff
@@ -217,6 +217,19 @@ def test_find(self):
                 if loc != -1:
                     self.assertEqual(i[loc:loc+len(j)], j)
 
+        # test tuple arguments
+        self.checkequal(2, '__aa__bb__', 'find', ('aa', 'bb'))
+        self.checkequal(2, '__aa__bb__', 'find', ('bb', 'aa'))
+        self.checkequal(-1, '__aa__bb__', 'find', ('cc', 'dd'))
+        self.checkequal(-1, '__aa__bb__', 'find', ())
+        self.checkequal(6, '__aa__bb__', 'find', ('aa', 'bb'), 3)
+        self.checkequal(-1, '__aa__bb__', 'find', ('aa', 'cc'), 3)
+        self.checkequal(2, '__aa__bb__', 'find', ('aa', 'bb'), 0, 10)
+        self.checkequal(-1, '__aa__bb__', 'find', ('aa', 'bb'), 0, 3)
+        self.checkequal(2, '__aa__bb__', 'find', ('aa', 'bb'), 0, 4)
+        s = '_' * 9998 + 'aaaa' + '_' * 9998
+        self.checkequal(9998, s, 'find', ('aaaa', 'bb'))
+
```

Comment:
```suggestion
1. @methane 's example for priority check can be useful
Think about "foobar".find(("ob", "oba")) and "foobar".find(("ob", "oob"))

"foobar".find("ob") => result = 2
cur_end = result + sublen -1 = 2 + 3 -1 = 4
"foob".find("oba") => new_result = -1 # its OK because result is 2 already
"foob".find("oob") => new_result = 1
2. And empty-string substring case
```